python/tank/commands/setup_project.py

```diff
@@ -175,7 +175,7 @@ def run_noninteractive(self, log, parameters):
             log.info("Localizing Core...")
             core_localize.do_localize(
                 log,
-                self.tk.shotgun,
+                self._shotgun_connect(log),
```

Comment:
Make sure you can run setup_project without a Toolkit instance, i.e. just the API.